src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,107 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.subsystems;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.Constants;
+import frc.robot.constants.ElevatorConstants;
+import frc.robot.controls.Operator;
+import frc.robot.util.MotorFactory;
+
+public class Elevator extends SubsystemBase {
+
+  private final WPI_TalonFX m_elevatorMotor; //do I define these are private final? 
+  private final DigitalInput m_bottomLimitSwitch;
+  private final DigitalInput m_topLimitSwitch; 
+  private final PIDController m_elevatorPID; 
+  private final DutyCycleEncoder m_absoluteSpoolEncoder; 
+
+  public Elevator() {
+    this(
+      MotorFactory.createTalonFX(ElevatorConstants.kElevatorMotor, Constants.kRioCAN),
+      new DigitalInput(-1),
+      new DigitalInput(-1),
+      new DutyCycleEncoder(-1)
+    );
+  }
+
+  public Elevator(WPI_TalonFX motor, DigitalInput bottomSwitch, DigitalInput topSwitch, DutyCycleEncoder absoluteSpoolEncoder) {
+    m_elevatorMotor = motor; 
+    m_bottomLimitSwitch = bottomSwitch; 
+    m_topLimitSwitch = topSwitch;
+    m_absoluteSpoolEncoder = absoluteSpoolEncoder; 
+
+    m_elevatorPID = new PIDController(ElevatorConstants.kElevatorP, ElevatorConstants.kElevatorI, ElevatorConstants.kElevatorD);  
+  }
+
+  public void set(double power){
+    m_elevatorMotor.set(power); 
+  }
+
+  public void stopMotor(){
+    m_elevatorMotor.set(0); 
+  }
+
+  public boolean returnBottomLimSwitchCondition(){
+    return m_bottomLimitSwitch.get(); 
+  }
+
+  public boolean returnTopLimSwitchCondition(){
+    return m_topLimitSwitch.get(); 
+  }
+
+  public void resetMotorEncoder(){
+    m_elevatorMotor.setSelectedSensorPosition(0); 
+  }
+
+  public void stopMotorsIfLimitSwitchesTripped(double power){
```

Comment:
The two if cases are unnecessary. Also, I think this should just be the default behavior of the set() function. I would talk to joshua about the wiring of the limit switches (or hall effect sensors), and if they can be inputted into the falcon. That would allow you to set it up so the motor automatically does not power forward if the forward limit switch is enabled. See here for documentation: https://v5.docs.ctr-electronics.com/en/latest/ch13_MC.html#limit-switches

Reply:
Code Change added.

Reply:
Sorry, are you sure you pushed? The function is still there.

Reply:
I kept it there so that we can try both using the encoders plugged into the Rio and the Falcon

Reply:
Well, to do that, you would need to change a different part of the code, so I'm not sure this function is necessary no matter where you plug the limit switches into.

Reply:
Hi Anthony, I deleted functionality for using limit switches plugged into the RoboRio